docs/ocp-deployment.md

```diff
@@ -0,0 +1,844 @@
+
+This guide shows how to deploy the Inferno-Autoscaler (integrated with the HorizontalPodAutoscaler) on OpenShift (OCP), allowing vLLM servers to scale accordingly to the observed load.
+
+**Note**: the deployment was tested on an OCP cluster with *admin* privileges.
+
+## Overview
+
+After deploying the Inferno-autoscaler following the provided guides, this guide allows the integration of the following components:
+
+1. **Inferno Controller** processes VariantAutoscaling objects and emits the `inferno_desired_replicas` metrics
+
+2. **Prometheus and Thanos** scrape these metrics from the Inferno-autoscaler `/metrics` endpoint using TLS, using the existing monitoring infrastructure present on OCP
+
+3. **Prometheus Adapter** exposes the metrics to Kubernetes external metrics API
+
+4. **HPA** reads the value for the `inferno_desired_replicas` metrics and adjusts Deployment replicas accordingly
+
+5. **llm-d infrastructure**: a Gateway and the Inference Scheduler are deployed to forward requests to the pods running vLLM instances.
+
+## Prerequisites
+
+- Prometheus stack already present on the OCP cluster
+- All components must be fully ready before proceeding: 2-3 minutes may be needed after the deployment
+
+### 0. Deploy the Inferno-Autoscaler and llm-d
+
+First, export the following environment variables:
+
+```sh
+export WVA_PROJECT=$PWD
+export BASE_NAME="inference-scheduling"
+export NAMESPACE="llm-d-$BASE_NAME"
+export EXAMPLES_DIR="examples/$BASE_NAME"
+export MONITORING_NAMESPACE="openshift-user-workload-monitoring" 
+```
+
+*Note*: the `MONITORING_NAMESPACE` environment variable may need to change depending on your OCP setup.
+
+#### Deploying the Inferno-Autoscaler
+
+Before running the Make target to deploy Inferno-Autoscaler, the `PROMETHEUS_BASE_URL` in the `config/manager/configmap.yaml` must be changed into a valid URL, to be able to connect to Thanos:
+
+**Note**: the Prometheus/Thanos URL may change depending on your OCP setup.
+
+```yaml
+# ...
+  PROMETHEUS_BASE_URL: "https://thanos-querier.openshift-monitoring.svc.cluster.local:9091"
+```
+
+After that, you can deploy the Inferno-Autoscaler using the basic `Make` target:
+
+```sh
+make deploy IMG=quay.io/infernoautoscaler/inferno-controller:0.0.1-multi-arch
+```
+
+Then, you need to deploy the required ConfigMaps for the accelerator costs and the service classes. An example of this configuration can be found in the following command.
+
+```sh
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+# This configMap defines the set of accelerators available
+# to the autoscaler to assign to variants
+#
+# For each accelerator, need to specify a (unique) name and some properties:
+# - device is the name of the device (card) corresponding to this accelerator,
+#   it should be the same as the device specified in the node object
+# - cost is the cents/hour cost of this accelerator
+#
+metadata:
+  name: accelerator-unit-costs
+  namespace: inferno-autoscaler-system
```

Comment:
is this correct? I get an error - does not exist. Should this use the NS defined in the export above?